action.yml

```diff
@@ -22,6 +22,10 @@ inputs:
     description: 'Fail the workflow if cache entry is not found'
     default: 'false'
     required: false
+  lookup-only:
+    description: 'Skip downloading cache. Only check if cache entry exists'
```

Comment:
Does it check for exact match only? If so, then `restore-keys` will be ignored. May be we should call that out here.

Reply:
`restore-keys` won't be ignored, we can update the description to something like

```yaml
description: 'Check if a cache entry exists for the given input(s) (key, restore-keys) without downloading the cache'
```

Reply:
@kotewar @cdce8p what is the output when restore key matches? We can't set `cache-hit` to true as that is true only in case of exact match.

Reply:
The output `cache-hit` will be false in that case.

Reply:
> If I understand correctly there is currently no difference in output between no exact match but a partial match.

Not quite. `cache-hit` will be `true` for exact matches. Think of it like this: When adding `lookup-only` the action behaves just the same (incl. the outputs) as without the argument. The only difference being that the cache entry isn't actually downloaded and extracted.

Reply:
I see. ~Normally there is a purpose of using the `restore-keys` because you are actually restoring something, but `cache-hit` will still be false. So from the view of the action there is still a cache miss. With this option there are no side effects when a match has occurred on one of the `restore-keys`.~

While writing this I'm wondering: did I always misinterpret the note on `cache-hit` and does it also return `true` when an exact hit has been found on one of the `restore-keys` but just not on a partial match on the `restore-keys`?

Reply:
> While writing this I'm wondering: did I always misinterpret the note on `cache-hit` and does it also return `true` when an exact hit has been found on one of the `restore-keys` but just not on a partial match on the `restore-keys`?

`cache-hit` will only be `true` for an exact match on the primary key.
https://github.com/actions/cache/blob/940f3d7cf195ba83374c77632d1e2cbb2f24ae68/src/restoreImpl.ts#L67-L72

That doesn't mean though there aren't no side effects now with `lookup-only: true` if the key doesn't match. `cache-matched-key` is still set.
https://github.com/actions/cache/blob/940f3d7cf195ba83374c77632d1e2cbb2f24ae68/src/restoreImpl.ts#L64-L65

--
In practice, I suspect the option will most likely only be used to check for an exact key (i.e. without `restore-keys`) but there might be other use cases I haven't though about yet which could use it. In any way as explained here https://github.com/actions/cache/pull/1041#issuecomment-1409467933 the current implementation is not only simpler then a dedicated `check` action #1095 (which would have provided the option to remove the input) but also provides a better UX.

Reply:
Thank you. If that's always the case, then `cache-matched-key` should be mentioned as an output of the main action. Currently the only documented output is `cache-hit`. Therefore I assumed there were no side-effects in that case. Or this option should only be available in the restore action where this output is actually documented.

> -- In practice, I suspect the option will most likely only be used to check for an exact key (i.e. without `restore-keys`) but there might be other use cases I haven't though about yet which could use it. In any way as explained here [#1041 (comment)](https://github.com/actions/cache/pull/1041#issuecomment-1409467933) the current implementation is not only simpler then a dedicated `check` action #1095 (which would have provided the option to remove the input) but also provides a better UX.

I currently have a usecase where I cache all images in a matrix, and use them in another different matrix later on. In the setup I assume that if one file is successfully cached, all files (with the same prefix) are successfully cached. If this assumption is incorrect the matrix workflow will still succeed but less efficient (however it is unlikely that it would ever happen). It would be a waste to spin up the entire caching matrix just to check each key for a match. During the matrix workflow, each job uses only one file so there is no need to cache them into one key.

But in this case I can check for a match on one particular key instead and it will still have the same result.

Reply:
I agree, `cache-matched-key` (and even `cache-primary-key` like we have in `restore` action) will be helpful in the main action's output. We can plan to add it. In case anyone wants to contribute for the same, please feel free to do so. 😄